pkg/pool-utils/contracts/ProtocolFeeCache.sol

```diff
@@ -126,6 +128,13 @@ abstract contract ProtocolFeeCache is RecoveryMode {
         _updateProtocolFeeCache(_protocolFeeProvider, ProtocolFeeType.AUM);
     }
 
+    /**
+     * @dev Override in derived contracts to perform some action before the cache is updated. This is typically relevant
+     * in Pools that have protocol debt that is dependent in the protocol fee percentages, as it needs to be paid out
+     * before said percentages change.
```

Comment:
```suggestion
     * @dev Override in derived contracts to perform some action before the cache is updated. This is typically relevant
     * to Pools that incur protocol debt between operations. To avoid altering the amount due retroactively,
     * this debt needs to be paid before the fee percentages change.
```
Protocol debt would always be dependent on the fee percentages.